app/src/main/java/com/example/android/codelabs/navigation/mvp/view/MainActivityView.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.android.codelabs.navigation.mvp.view
+
+import android.content.res.Resources
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import androidx.navigation.Navigation.findNavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.NavigationUI.setupActionBarWithNavController
+import androidx.navigation.ui.navigateUp
+import androidx.navigation.ui.onNavDestinationSelected
+import androidx.navigation.ui.setupWithNavController
+import com.example.android.codelabs.navigation.R
+import com.example.android.codelabs.navigation.mvp.contracts.MainActivityContracts
+import kotlinx.android.synthetic.main.navigation_activity.bottom_nav_view
+import kotlinx.android.synthetic.main.navigation_activity.drawer_layout
+import kotlinx.android.synthetic.main.navigation_activity.nav_view
+import kotlinx.android.synthetic.main.navigation_activity.toolbar
+
+class MainActivityView : MainActivityContracts.View {
+    private lateinit var activity: AppCompatActivity
```

Comment:
use a weakReference instead of the AppCompatActivity variable pls

Reply:
you can use the activityBaseView from the other projects

Reply:
done